src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,69 @@
 package core.basesyntax;
 
+import static java.lang.Integer.parseInt;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    public static final String REGEX = " ";
+    public static final String BUY = "buy";
+    public static final String SUPPLY = "supply";
+    public static final String RESULT_REGEX = ",";
+    public static final String RESULT_NAME = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        writeDataToFile(getFormattedData(readDataFromFile(fromFileName)), toFileName);
+    }
+
+    private String readDataFromFile(String fromFileName) {
+        File readFile = new File(fromFileName);
+
+        StringBuilder builder = new StringBuilder();
+        try (BufferedReader reader = new BufferedReader(new FileReader(readFile))) {
+            while (reader.ready()) {
+                builder.append(reader.readLine()).append(REGEX);
+            }
+        } catch (IOException exception) {
+            throw new RuntimeException("Can't read data from file " + readFile, exception);
+        }
+        return builder.toString();
+    }
+
+    private String getFormattedData(String data) {
+        int result;
+        int totalSupply = 0;
+        int totalBuy = 0;
+        StringBuilder builder = new StringBuilder();
+        String[] arr = data.split(REGEX);
+        for (String tips : arr) {
+            final int intValue = parseInt(tips.substring(tips.indexOf(RESULT_REGEX) + 1));
```

Comment:
[use split instead of substring](https://mate-academy.github.io/jv-program-common-mistakes/java-core/builder-file/work-with-file#use-split-instead-of-substring)